edk2toolext/invocables/edk2_update.py

```diff
@@ -65,12 +108,58 @@ def GetVerifyCheckRequired(self):
         return False
 
     def GetSettingsClass(self):
-        '''  Providing UpddateSettingsManger '''
+        '''  Providing UpdateSettingsManger '''
         return UpdateSettingsManager
 
     def GetLoggingFileName(self, loggerType):
         return "UPDATE_LOG"
 
+    def AddCommandLineOptions(self, parserObj):
+        ''' adds command line options to the argparser '''
+        # This will parse the packages that we are going to update
+        parserObj.add_argument('-p', '--pkg', '--pkg-dir', dest='packageList', type=str,
```

Comment:
Additionally, we could move this out to some sort of common file for Target/Package/Arch and pass in the ParserObj to get these added

Reply:
created a new intermediate base class.  Looks better. thanks for the feedback